spacewalk/config/etc/httpd/conf.d/zz-spacewalk-www.conf

```diff
@@ -51,9 +51,8 @@ RewriteRule ^/index\.html$ /rhn/manager/login
 # For rhn-custom-info
 RewriteRule ^/WEBRPC /rhn/rpc/api
 
-# Rewrite rules for APT repos metadata
-RewriteRule ^/rhn/manager/download/dists/(.*)/main/(.*)/(Packages.*)$ /rhn/manager/download/$1/repodata/$3
-RewriteRule ^/rhn/manager/download/dists/(.*)/(InRelease|Release.*)$ /rhn/manager/download/$1/repodata/$2
+# Rewrite rule for APT repos metadata
+RewriteRule ^/rhn/manager/download/(.*)/(InRelease|Release.*|Packages.*)$ /rhn/manager/download/$1/repodata/$2
```

Comment:
I see problems with the upgrade.
After you update the server, all debian like clients are broken as dist/* is not redirected anymore.
You need to regenerate all the metadata and force all clients to re-write the new style channel list .
I think this can be done via "highstate", but it is not enforced by the update.

What would happen if a client want to access a repository without the changes from `channels.sls` are applied?

Reply:
Release files should be regenerated. I've implemented it with https://github.com/uyuni-project/uyuni/pull/4459/commits/eefe6564136ef9b709c339eda8903fb5c82e6bce
`channels` state is applied automatically on package/patch installation with web UI, if I'm not mistaken. Package/patch installation with salt directly will fail if `channels` state not applied before. Is it possible to force apply this state only?

Reply:
not automatically. If you use the UI and go to the channels page (or SSM Channels) and you just click through it without changing anything you get this done. But it is still a manual step.
But if it works on just updating a package it might be sufficient. Only the people using apt-get directly on the client step into it.

Reply:
We can left the deleted strings in the config for a while and delete them on next release or point in the documentation that applying `channels` state is required for all debian based systems after installing the update.

Reply:
If Silvio is fine with this, I am ok as well.

Reply:
Yeah, unless it creates problems please leave old redirections intact for a while.

My only ask would be to add a `# TEMPORARY: <explanation on when following lines can be removed>` comment for posterity.

Reply:
@moio I think it's the best option for such change.